crates/autopilot/src/run_loop.rs

```diff
@@ -908,6 +915,11 @@ struct Metrics {
     /// Tracks the last executed auction.
     auction: prometheus::IntGauge,
 
+    /// Tracks the number of winners per auction.
+    /// Label value is the exact number of winners.
+    #[metric(labels("num_winners"))]
+    auction_winners: prometheus::IntCounterVec,
```

Comment:
Seems like you are using an `IntCounterVec` like a `Histogram`. Usually the labels of the counters are supposed to be specific items (e.g. solvers, error kinds, etc.) - here you are using them as buckets.
When you switch to using `Histogram` make sure to also define reasonable buckets with `buckets(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)` as the defaults aren't great for counting purposes.